engine/src/cmd/basecomputer.cpp

```diff
@@ -2717,10 +2717,10 @@ bool BaseComputer::sellSelectedCargo( int requestedQuantity )
         Cargo     sold;
         const int quantity = (requestedQuantity <= 0 ? item->quantity : requestedQuantity);
         if (item->mission) {
-            vector< Cargo >::iterator mycargo = std::find( playerUnit->pImage->cargo.begin(),
-                                                           playerUnit->pImage->cargo.end(), *item );
-            if ( mycargo != playerUnit->pImage->cargo.end() )
-                playerUnit->RemoveCargo( mycargo-playerUnit->pImage->cargo.begin(), quantity, true );
+            vector< Cargo >::iterator mycargo = std::find( playerUnit->cargo.begin(),
+                                                           playerUnit->cargo.end(), *item );
+            if ( mycargo != playerUnit->cargo.end() )
+                playerUnit->RemoveCargo( mycargo-playerUnit->cargo.begin(), quantity, true );
```

Comment:
Dropping the pImage layer should be fine/a good move assuming a 1:1 relationship.

Reply:
For a moment there, I had serious doubts.
But I don't see how it would compile if it weren't.

Reply:
if `pImage` was getting moved between different images for the same object, then it might create an inconsistency that won't be easily obvious; depends on what `pImage` is doing.

Reply:
It started as am image to be displayed in the targeting hud. Then stuff like this migrated there for some reason.